### PR TITLE
Fix bad credentials error not displayed

### LINKED ISSUE #267: Bad credential not raised for these connectors
No call to next('bad credentials')
- [ ] ./googlecontacts.coffee
- [ ] ./sncf.js
- [ ] ./isen.coffee
- [ ] ./birthdays.js
- [ ] ./linkedin.js
- [ ] ./ovh_eu.coffee
- [ ] ./ovh_ca.coffee
- [ ] ./soyoustart_eu.coffee
- [ ] ./soyoustart_ca.coffee
- [ ] ./kimsufi_eu.coffee
- [ ] ./kimsufi_ca.coffee
- [ ] ./ical_feed.js
- [ ] ./github.coffee
- [ ] ./sfr_box.js
- [ ] ./sfr_mobile.js
- [ ] ./nest.coffee
- [ ] ./online_net.coffee
- [ ] ./runabove.coffee
- [ ] ./rescuetime.coffee
- [ ] ./digitalocean.coffee
- [ ] ./bouyguesbox.coffee
- [ ] ./bouyguestelecom.coffee
- [ ] ./orange.coffee
- [ ] ./sosh.coffee

### PR DESCRIPTION
**Pull request by nicofrand**
Fix #267 

- [x] ./googlecontacts.coffee
- [x] ./sncf.js
- [x] ./isen.coffee
- [ ] ~~./birthdays.js~~ not needed
- [ ] ~~./linkedin.js~~ won't fix, will fix when rewriting the connector
- [x] ./ovh_eu.coffee
- [x] ./ovh_ca.coffee
- [x] ./soyoustart_eu.coffee
- [x] ./soyoustart_ca.coffee
- [x] ./kimsufi_eu.coffee
- [x] ./kimsufi_ca.coffee
- [x] ./ical_feed.js
- [x] ./github.coffee
- [x] ./sfr_box.js
- [x] ./sfr_mobile.js
- [ ] ~~./nest.coffee~~ the API does not seem to allow to handle errors
- [x] ./online_net.coffee
- [x] ./runabove.coffee
- [x] ./rescuetime.coffee
- [x] ./digitalocean.coffee
- [x] ./bouyguesbox.coffee
- [x] ./bouyguestelecom.coffee
- [x] ./orange.coffee
- [x] ./sosh.coffee

### CODE REVIEW

**Comment by nicofrand** [comment]
This needs a review, @ZeHiro, @doubleface, if you have some time…

File: server/konnectors/orange.coffee [hunk 1]
```diff
@@ -94,34 +94,36 @@ logIn = (requiredFields, billInfos, data, next) ->
...
-                        else
-                            log.info 'Fetch bill info succeeded'
-                            data.html = body
-                            next()
+                return next 'request error'
+
+            response = JSON.parse body
+            if response.credential? or response.password?
+                error = if response.credential? then response.credential
+                else response.password
```
Comment (ZeHiro) [inline server/konnectors/orange.coffee]: @nicofrand : the `else` is interpretted correctly?

File: server/konnectors/sosh.coffee [hunk 2]
```diff
@@ -95,28 +95,29 @@ logIn = (requiredFields, billInfos, data, next) ->
...
+            log.info 'Fetch bill info'
+            request billOptions, (err, res, body) ->
+                if err
+                    log.error 'An error occured while fetching bills'
+                    console.log err
+                    return next 'request error'
+
+                log.info 'Fetch bill info succeeded'
+                data.html = body
+                next()
```
Comment (ZeHiro) [inline server/konnectors/sosh.coffee]: return next()

File: server/konnectors/rescuetime.coffee [hunk 3]
```diff
@@ -111,7 +113,7 @@ Something went wrong while fetching rescue time data.
                         log.debug 'new activity imported'
                         log.debug JSON.stringify data
 
-                        cb err
+                        cb()
```
Comment (ZeHiro) [inline server/konnectors/rescuetime.coffee]: return cb()

File: server/konnectors/ical_feed.js [hunk 4]
```diff
@@ -41,11 +41,12 @@ function downloadFile(requiredFields, entries, data, next) {
...
-    } else {
-      connector.logger.info('Download succeeded.');
-      data.ical = body;
+      return next('request error');
     }
-    next(err);
+
+    connector.logger.info('Download succeeded.');
+    data.ical = body;
+    next();
```
Comment (ZeHiro) [inline server/konnectors/ical_feed.js]: missing return

File: server/konnectors/orange.coffee (hunk 1 above)
Comment (nicofrand) [inline server/konnectors/orange.coffee]: @ZeHiro yep : https://github.com/cozy-labs/konnectors/blob/master/build/server/konnectors/orange.js#L105

**Pull request merged**